File: Java/src/main/java/Solution101_200/Solution169.java
package Solution101_200;

/**
 * @author windows
 */
public class Solution169 {
    public int majorityElement(int[] nums) {
        int count = 1;
        int people = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == people){
                count++;
            }else{
                count--;
                if (count == 0){
                    people = nums[i];
                    count = 1;
                }
            }
        }
        return people;
    }


    public static void main(String[] args) {
        Solution169 solution169 = new Solution169();
    }
}

File: C++/src/Solution101_200/Solution169.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution169.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;


int Solution169::majorityElement(vector<int> &nums) {
    int num = nums[0];
    int ticket = 1;
    for (int i = 1; i < nums.size(); ++i) {
        if (nums[i] == num) {
            ticket++;
        }else{
            if(--ticket == 0){
                num = nums[i];
                ticket = 1;
            }
        }
    }
    return num;
}



